lib/file-view.coffee

```diff
@@ -19,7 +19,11 @@ class FileView extends HTMLElement
     @fileName.dataset.name = @file.name
     @fileName.dataset.path = @file.path
 
-    @fileName.classList.add(FileIcons.getService().iconClassForPath(@file.path))
+    iconClass = FileIcons.getService().iconClassForPath(@file.path, this)
```

Comment:
Thing is, I want to avoid extra filesystem calls if I can help it. This information is already accessible on the object at the time of the request... I agree that passing state around can introduce potential complications, but I still feel extra metadata about the file itself should be supplied.

Damn well agreed with the `Show icons in {tabs/tree-view}`, though.